flow/shared/string.go

```diff
@@ -25,3 +29,7 @@ func ReplaceIllegalCharactersWithUnderscores(s string) string {
 func IsValidReplicationName(s string) bool {
 	return reLegalIdentifierLower.MatchString(s)
 }
+
+func JoinHostPort[I constraints.Integer](host string, port I) string {
```

Comment:
nit: instead of importing an `x/exp` package can we define our own constraint

Reply:
perhaps, but we already use `x/exp` elsewhere so this isn't a new dependency